app/src/main/java/com/example/bigowlapp/utils/AuthenticatorByDeviceId.java

```diff
@@ -0,0 +1,49 @@
+package com.example.bigowlapp.utils;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Attendance;
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.model.UserScheduleResponse;
+import com.example.bigowlapp.repository.RepositoryFacade;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class AuthenticatorByDeviceId {
+
+    private String deviceIdNumber;
+    private final Context context;
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+    RepositoryFacade repositoryFacade = RepositoryFacade.getInstance();
+
+    public AuthenticatorByDeviceId(Context context) {
+        this.context = context;
+    }
+
+    @SuppressLint("MissingPermission")
```

Comment:
Again, don't suppress lint and add the permission of READ_PRIVILEGED_PHONE_STATE to androidManifest. However, I tried fixing this, but seems like you have to keep @SuppressLint("HardwareIds") instead. Try to fix it and find the best safe way

Reply:
I put a  comment to explain that the permission was already granted by  the user as a condition to proceed after the sign in step